cardano-node/src/Cardano/Node/Configuration/TopologyP2P.hs

```diff
@@ -0,0 +1,201 @@
+  )
+where
+
+import           Cardano.Prelude
+import           Prelude (String)
+
+import qualified Control.Exception as Exception
+import           Data.Aeson
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy.Char8 as LBS
+import qualified Data.Text as Text
+
+import           Cardano.Node.Configuration.POM (NodeConfiguration (..))
+import           Cardano.Slotting.Slot (SlotNo (..))
+
+import           Cardano.Node.Types
+import           Cardano.Node.Configuration.Topology (TopologyError (..))
+
+import           Ouroboros.Network.NodeToNode (PeerAdvertise (..))
+import           Ouroboros.Network.PeerSelection.LedgerPeers (UseLedgerAfter (..))
+import           Ouroboros.Network.PeerSelection.RelayAccessPoint (RelayAccessPoint (..))
+
+newtype UseLedger = UseLedger UseLedgerAfter deriving (Eq, Show)
+
+instance FromJSON UseLedger where
+  parseJSON (Data.Aeson.Number n) =
+    if n >= 0 then return $ UseLedger $ UseLedgerAfter $ SlotNo $ floor n
+              else return $ UseLedger   DontUseLedger
+  parseJSON _ = mzero
+
+instance ToJSON UseLedger where
+  toJSON (UseLedger (UseLedgerAfter (SlotNo n))) = Number $ fromIntegral n
+  toJSON (UseLedger DontUseLedger)               = Number (-1)
+
+data NodeSetup = NodeSetup
+  { nodeId :: !Word64
+  , nodeIPv4Address :: !(Maybe NodeIPv4Address)
+  , nodeIPv6Address :: !(Maybe NodeIPv6Address)
+  , producers :: ![RootAddress]
+  , useLedger :: !UseLedger
+  } deriving (Eq, Show)
+
+instance FromJSON NodeSetup where
+  parseJSON = withObject "NodeSetup" $ \o ->
+                NodeSetup
+                  <$> o .:  "nodeId"
+                  <*> o .:  "nodeIPv4Address"
+                  <*> o .:  "nodeIPv6Address"
+                  <*> o .:  "producers"
+                  <*> o .:? "useLedgerAfterSlot" .!= UseLedger DontUseLedger
+
+instance ToJSON NodeSetup where
+  toJSON ns =
+    object
+      [ "nodeId"             .= nodeId ns
+      , "nodeIPv4Address"    .= nodeIPv4Address ns
+      , "nodeIPv6Address"    .= nodeIPv6Address ns
+      , "producers"          .= producers ns
+      , "useLedgerAfterSlot" .= useLedger ns
+      ]
+
+data RootAddress = RootAddress
+  { addrs :: [RelayAccessPoint]
+  , advertise :: PeerAdvertise
+  } deriving (Eq, Show)
+
+instance FromJSON RootAddress where
+  parseJSON = withObject "RootAddress" $ \o ->
+                RootAddress
+                  <$> o .:  "addrs"
+                  <*> o .:? "advertise" .!= DoNotAdvertisePeer
+
+instance ToJSON RootAddress where
+  toJSON ra =
+    object
+      [ "addrs" .= addrs ra
+      , "advertise" .= advertise ra
+      ]
+
+-- | Transforms a 'RootAddress' into a pair of 'RelayAccessPoint' and its
+-- corresponding 'PeerAdvertise' value.
+--
+rootAddressToRelayAccessPoint
+  :: RootAddress
+  -> [(RelayAccessPoint, PeerAdvertise)]
+rootAddressToRelayAccessPoint RootAddress { addrs, advertise } =
+    [ (ra, advertise) | ra <- addrs ]
+
+data LocalRootPeers = LocalRootPeers
```

Comment:
@karknu I think we should rename this type to `LocalRootPeerGroup`, as this configures a single local root group.  This would also fit with the other name `LocalRootPeersGroups ~ [LocalRootPeerGroup]`.  Do you agree?